packages/stack/src/lib/cookie.ts

```diff
@@ -30,37 +26,38 @@ export function setOrDeleteCookie(name: string, value: string | null, options: S
 }
 
 export function deleteCookie(name: string) {
-  try {
+  if (typeof window !== "undefined") {
+    Cookies.remove(name);
+  } else {
     rscCookies().delete(name);
-  } catch (e: any) {
-    if (isRscCookieUnavailableError(e)) {
-      Cookies.remove(name);
-    } else {
-      throw e;
-    }
   }
 }
 
 export function setCookie(name: string, value: string, options: SetCookieOptions = {}) {
-  const isProd = process.env.NODE_ENV === "production";
-  try {
+  // Check if the current page is server over HTTPS with:
+  // 1. Check if the https cookie is set on the client (see comment above)
+  // 2. If on the client, check the protocol directly
+  // 3. If on the server, check the X-Forwarded-Proto header
```

Comment:
```suggestion
  // Whenever the client is on HTTPS, we want to set the Secure flag on the cookie.
  //
  // This is not easy to find out on a Next.js server, so we use the following steps:
  //
  // 1. If we're on the client, we can check window.location.protocol which is the ground-truth
  // 2. Check whether the stack-is-https cookie exists. This cookie is set in various places on
  //      the client if the protocol is known to be HTTPS
  // 3. Check the X-Forwarded-Proto header
  // 4. Otherwise, assume HTTP without the S
  //
  // Note that malicious clients could theoretically manipulate the `stack-is-https` cookie or
  // the `X-Forwarded-Proto` header; that wouldn't cause any trouble except for themselves,
  // though.
```